src/main/haskell/kore/src/Kore/Step/Step.hs

```diff
@@ -178,3 +305,71 @@ heatingCooling axiomStrategy axioms =
     normal = axiomStrategy normalRules
     coolingRules = filter isCoolingRule axioms
     cool = axiomStrategy coolingRules
+
+{-| A strategy for doing the first step of a one-path verification.
+For subsequent steps, use 'onePathFollowupStep'.
+
+It first removes the destination from the input, then it tries to apply
+the normal axioms.
+
+Whenever it applies an axiom, the subsequent axioms see only the part of the
+pattern to which the axiom wasn't applied.
+-}
+onePathFirstStep
+    :: axiom
+    -- ^ axiom with which to remove the destination we`re trying to reach
+    -> [axiom]
+    -- ^ normal axioms
+    -> Strategy (Prim axiom)
+onePathFirstStep destinationRemovalAxiom axioms =
+    Strategy.sequence
+        (  [axiomWithRemainderStep destinationRemovalAxiom]
+        ++ map
```

Comment:
Because each is applied on the previous axiom's remainder.

Reply:
That somewhat abuses what `Strategy.sequence` means, which is: that each of the configurations is a child of the configuration reached by applying the preceding axioms. The remainder pattern is not a child configuration, but exists in parallel (at the same execution step; I understand that `Strategy.parallel` is not what you need here).

Reply:
I see. Here's how I was interpreting it at the strategy level, which does not know anything about axioms: we have a primitive strategy that produces two results (the rewrite and the remainder), and we follow that with a strategy that works only on the remainder. By definition, this is a sequence. Of course, it remains true that the remainder is not a true child, but we could change the way we think about strategies.

There are two other options. First, we could not use strategies. I'm not that happy with this, but it's a possibility.

Second, we could change Strategy.hs so that the patterns returned by a strategy are tagged as either as `Result` or as `Remainder` and we could have something like `daisyChain` as an alternative to `sequence`. Would we gain enough to do this, or is it better to keep strategies simple?